connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointTask.java

```diff
@@ -190,4 +227,103 @@ public void commitRecord(SourceRecord record) {
             Checkpoint.unwrapGroup(record.sourcePartition()),
             System.currentTimeMillis() - record.timestamp());
     }
+
+    private void refreshIdleConsumerGroupOffset() {
+        Map<String, KafkaFuture<ConsumerGroupDescription>> consumerGroupsDesc = targetAdminClient
+            .describeConsumerGroups(consumerGroups).describedGroups();
+
+        for (String group : consumerGroups) {
+            try {
+                if (consumerGroupsDesc.get(group) == null) {
+                    // if consumerGroupsDesc does not contain this group, it should be the new consumer
+                    // group created at source cluster and its offsets should be sync-ed to target
+                    newConsumerGroup.add(group);
+                    continue;
+                }
+                ConsumerGroupDescription consumerGroupDesc = consumerGroupsDesc.get(group).get();
+                // sync offset to the target cluster only if the state of current consumer group is idle or dead
+                ConsumerGroupState consumerGroupState = consumerGroupDesc.state();
+                if (consumerGroupState.equals(ConsumerGroupState.EMPTY) || consumerGroupState.equals(ConsumerGroupState.DEAD)) {
+                    idleConsumerGroupsOffset.put(group, targetAdminClient.listConsumerGroupOffsets(group)
+                        .partitionsToOffsetAndMetadata().get().entrySet());
+                }
+            } catch (InterruptedException | ExecutionException e) {
+                log.error("Error querying for consumer group {} on cluster {}.", group, targetClusterAlias, e);
+            }
+        }
+    }
+
+    Map<String, Map<TopicPartition, OffsetAndMetadata>> syncGroupOffset() {
+        Map<String, Map<TopicPartition, OffsetAndMetadata>> offsetToSyncAll = new HashMap<>();
+
+        // first, sync offsets for the idle consumers at target
+        for (Map.Entry<String, Set<Map.Entry<TopicPartition, OffsetAndMetadata>>> group : idleConsumerGroupsOffset.entrySet()) {
+            String consumerGroupId = group.getKey();
+            // for each idle consumer at target, read the checkpoints (converted upstream offset)
+            // from the pre-populated map
+            Map<TopicPartition, OffsetAndMetadata> convertedUpstreamOffset = getConvertedUpstreamOffset(consumerGroupId);
+
+            if (convertedUpstreamOffset == null) continue;
+
+            Map<TopicPartition, OffsetAndMetadata> offsetToSync = new HashMap<>();
+            for (Entry<TopicPartition, OffsetAndMetadata> entry : group.getValue()) {
```

Comment:
I also think there might be an issue here. These are entries from the target cluster. So if the CG from the source cluster starts consuming from new Topics, we wouldn't translate it to the new cluster right?

Reply:
at line 297: https://github.com/apache/kafka/pull/7577/files#diff-5823cb84c328c779b8d18ab802109141R297 I added to sync the GC offset for the new consumers that appear at source cluster.

Also just added a log line after line 297 to help debug if there is actually a new GC and its offsets will be synced.

Reply:
@thspinto I think i am running into the same issue which you pointed here , so the source consumer group has different topic and is active and the target consumer group is idle but having different topic, but the code only checks for the topics and partitions matching the target site and adds them only when the target offset is less than source, it ignores other topics at the source

Reply:
If I am understanding right, are you asking about this scenario: consumer A is consuming from Topic `x` and ` y` and MM is replicating the offset of consumer A for Topic `x` and `y`. What if consumer A starts consume from Topic `x`, `y` and `z` where `z` is a new topic, why MM does not replicate the offset of consumer A for Topic `z`?

Reply:
Yes